tests/test_general.py

```diff
@@ -36,71 +51,75 @@ def test_general_init():
             assert h[{0: idx, 1: idx}] == 0
             assert h[{"x": idx, "y": idx}] == 0
 
-    # with named axes
-    assert Hist(
+    assert unnamed_hist(
         axis.Regular(50, -3, 3, name="x"), axis.Regular(50, -3, 3, name="y")
     ).fill(np.random.randn(10), np.random.randn(10))
 
-    assert Hist(axis.Boolean(name="x"), axis.Boolean(name="y")).fill(
+    assert unnamed_hist(axis.Boolean(name="x"), axis.Boolean(name="y")).fill(
         [True, False, True], [True, False, True]
     )
 
-    assert Hist(
+    assert unnamed_hist(
         axis.Variable(range(-3, 3), name="x"), axis.Variable(range(-3, 3), name="y")
     ).fill(np.random.randn(10), np.random.randn(10))
 
-    assert Hist(axis.Integer(-3, 3, name="x"), axis.Integer(-3, 3, name="y")).fill(
-        np.random.randn(10), np.random.randn(10)
-    )
+    assert unnamed_hist(
+        axis.Integer(-3, 3, name="x"), axis.Integer(-3, 3, name="y")
+    ).fill(np.random.randn(10), np.random.randn(10))
 
-    assert Hist(
+    assert unnamed_hist(
         axis.IntCategory(range(-3, 3), name="x"),
         axis.IntCategory(range(-3, 3), name="y"),
     ).fill(np.random.randn(10), np.random.randn(10))
 
-    assert Hist(
+    assert unnamed_hist(
         axis.StrCategory(["F", "T"], name="x"), axis.StrCategory("FT", name="y")
     ).fill(["T", "F", "T"], ["T", "F", "T"])
```

Comment:
@LovelyBuggies These really aren't unit tests. A unit test would check _just_ construction, then _just_ filling (and would assume construction works, obviously, since it has to construct to fill). Tests should be treated like code; you have to rework them and maintain them occasionally, so keeping repetition down, and keeping the tests focused on one thing at a time is important, just like in code. (Just some tips for the future, still very good test coverage here :) )

Reply:
PS: The reason this matters in this case: if they were really unit tests, I could have the named construction run with the `"named_hist"` fixture, which would put in Hist, BaseHist, and NamedHist; but I can't do that because they make axes with names but then fill without names.

Eventually, we should be able to have all tests broken into pieces; the axis-with-names parts would run with NamedHist and one or both of the Hist's, and then name-only and positional-is-okay tests would have a different or no parametrized fixture.